source/opt/ir_builder.h

```diff
@@ -151,6 +157,39 @@ class InstructionBuilder {
                             GetContext()->TakeNextId(), ops));
     return AddInstruction(std::move(construct));
   }
+  // Adds an unsigned int32 constant to the binary.
+  // The |value| parameter is the constant value to be added.
+  ir::Instruction* Add32BitUnsignedIntegerConstant(uint32_t value) {
+    return Add32BitConstantInteger<uint32_t>(value, false);
+  }
+
+  // Adds either a signed or unsigned 32 bit integer constant to the binary
+  // depedning on the |sign|. If |sign| is true then the value is added as a
+  // signed constant otherwise as an unsigned constant.
+  template <typename T>
+  ir::Instruction* Add32BitConstantInteger(T value, bool sign) {
+    if (sign) assert(value > 0);
```

Comment:
Yeah that should be !sign, it's because the T doesn't have to be unsigned so if you try and store a negative value with an unsigned spirv type it will just end up storing some huge number. Updated now